plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -111,7 +111,10 @@ public function save_to_session( $customer ) {
 				$data[ $session_key ] = (string) $session_value;
 			}
 		}
-		WC()->session->set( 'customer', $data );
+
+		if ( WC()->session ) {
+			WC()->session->set( 'customer', $data );
+		}
```

Comment:
🤔 I'm wondering if we should have a "doing it wrong" message or some other warning, rather than having it just fail silently here.

Another thought, would it make sense to bail at the top of the function if the session object is null, rather than doing the data processing (including some potential database calls) first?

Reply:
Good suggestions @coreymckrill thanks!
I pushed the changes.